tests/test_director.py

```diff
@@ -299,6 +329,29 @@ def test_05_register_ecu_serial(self):
     self.assertIn(secondary_serial, inventory.ecu_public_keys)
     self.assertEqual(
         keys_pub['secondary'], inventory.ecu_public_keys[secondary_serial])
+    self.assertEqual(
+        keys_pub['secondary'], inventory.get_ecu_public_key(secondary_serial))
+
+    # This should be empty, but should not raise an UnknownECU error now.
+    self.assertFalse(inventory.get_ecu_manifests(secondary_serial))
+    self.assertIsNone(inventory.get_last_ecu_manifest(secondary_serial))
+
+    # Try registering the Secondary again. Normally, we expect a Spoofing error
+    # as with the Primary, but for now, the registration attempt is simply
+    # ignored instead of raising an exception, for hacky demo reasons. /:
```

Comment:
Is the Secondary code documented to make it clear that although a hacky implementation is used, it will eventually change?  Perhaps a reason why a hacky implementation is in place might help for maintainability.

Reply:
The problem originates with the fact that the external web frontend code incorrectly performs ECU registrations every time it assigns a target to an ECU. I think I'll just fix that now. I'll give it a few minutes of work and see how that goes.

Reply:
Fixed.